src/cpp/preprocessor/aie2/aie2_blob_preprocessor_input.h

```diff
@@ -269,5 +270,69 @@ class aie2_asm_preprocessor_input : public aie2_blob_transaction_preprocessor_in
     aie2_blob_transaction_preprocessor_input::set_args(mc_code, patch_json, control_packet, libs, libpaths, ctrlpkt);
   }
 };
+
+class config_preprocessor_input : public aie2_blob_transaction_preprocessor_input
+{
+  uint32_t col = 0;
+protected:
+  void readconfigjson(std::istream& patch_json);
+  void add_pdi(const boost::property_tree::ptree& pinstance);
+  void add_instance(const boost::property_tree::ptree& pinstance);
+  class argument {
+    public:
+    std::string name;
+    std::string type;
+    std::string offset;
+
+    argument(const std::string& na, const std::string& ty, const std::string& off): name(na), type(ty), offset(off) {}
```

Comment:
pass by value and move is possibly more efficient for the compiler.
```suggestion
    argument(std::string na, std::string ty, std::string off)
    : name(std::move(na))
    , type(std::move(ty))
    , offset(std::move(off)) {}
```